src/Catalog/CatalogWriterBase.cs

```diff
@@ -183,7 +191,7 @@ protected virtual StorageContent CreateIndexContent(IGraph graph, Uri type)
         {
             IGraph graph = new Graph();
 
-            INode resourceNode = graph.CreateUriNode(resourceUri);
+            INode resourceNode = graph.CreateUriNode(CatalogItem.AliasesAddress(resourceUri, _customDomain));
```

Comment:
These changes seem very scary to me.

Basically, in the past, all had were resource uris with the same host, so we passed these uris around on their own. Now, however, we care about the host, and so we have added tons of places where we are replacing the host.

If we missed a place in which the host should have been replaced, we could very easily get into a scenario where some uri somewhere is being written with an incorrect host. Additionally, in the future, if we want to provide some additional modifications to the uri, we could potentially need to modify each of these places again, and could again miss.

I think we should rethink this design. I would much rather see something in which we stop passing around uris and start passing around a `ResourceIdentity` object that we can ask for the uri of which would perform the `AliasesAddress` transform. By changing interfaces and methods that previously used uris to `ResourceIdentity`s, we can guarantee through strong typing that the uris are being transformed when expected.

I understand doing that would probably be a much bigger change, but I think it would be a lot more stable in the long run.